sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetQuerySuite.scala

```diff
@@ -625,6 +625,22 @@ class ParquetQuerySuite extends QueryTest with ParquetTest with SharedSQLContext
       }
     }
   }
+
+  test("SPARK-15804: write out the metadata to parquet file") {
+    val data = (1, "abc") ::(2, "helloabcde") :: Nil
+    val df = spark.createDataFrame(data).toDF("a", "b")
+    val md = new MetadataBuilder().putString("key", "value").build()
+    val dfWithmeta = df.select(Column("a"), Column("b").as("b", md))
```

Comment:
Use `df.select('a, 'b.as("b", md))` instead. It's more compact and "modern". Our tests should promote the latest API.